test/Makefile

```diff
@@ -15,11 +18,14 @@ build:
 
 # Build the project with coverage instrumentation
 build-coverage:
-	CARGO_INCREMENTAL=0 RUSTFLAGS="$(RUSTFLAGS_COVERAGE)" cargo build $(FEATURES)
+	CARGO_INCREMENTAL=0 RUSTFLAGS="$(RUSTFLAGS_COVERAGE)" cargo +nightly build $(FEATURES)
 
 # Test the project
 test: build
```

Comment:
Can you include the `build-image` stage or something that builds the docker image automatically?